azure-devops/azext_devops/dev/team/service_endpoint.py

```diff
@@ -92,52 +84,86 @@ def create_service_endpoint(service_endpoint_type, authorization_scheme, name,
-            github_access_token = os.environ[AZ_DEVOPS_GITHUB_PAT_ENVKEY]
-
-        service_endpoint_authorization = EndpointAuthorization(
-            parameters={'accessToken': github_access_token},
-            scheme=SERVICE_ENDPOINT_AUTHORIZATION_PERSONAL_ACCESS_TOKEN)
-        service_endpoint_to_create = ServiceEndpoint(
-            authorization=service_endpoint_authorization,
-            name=name, type=SERVICE_ENDPOINT_TYPE_GITHUB, url=github_url)
-        return client.create_service_endpoint(service_endpoint_to_create, project)
 
-    if (service_endpoint_type == SERVICE_ENDPOINT_TYPE_AZURE_RM and
-            authorization_scheme == SERVICE_ENDPOINT_AUTHORIZATION_SERVICE_PRINCIPAL):
+    service_endpoint_authorization = EndpointAuthorization(
+        parameters={'tenantid': azure_rm_tenant_id,
+                    'serviceprincipalid': azure_rm_service_principal_id},
+        scheme=SERVICE_ENDPOINT_AUTHORIZATION_SERVICE_PRINCIPAL)
 
+    if azure_rm_service_principal_certificate_path is None:
         AZURE_RM_SP_KEY_END_VARIABLE_NAME = CLI_ENV_VARIABLE_PREFIX + 'AZURE_RM_SERVICE_PRINCIPAL_KEY'
         if AZURE_RM_SP_KEY_END_VARIABLE_NAME not in os.environ:
             error_message = 'Please specify azure service principal key in ' + AZURE_RM_SP_KEY_END_VARIABLE_NAME +\
-                            ' environment variable in non-interactive mode.'
+                            ' environment variable in non-interactive mode or use ' +\
+                            '--azure-rm-service-principal-certificate-path'
             verify_is_a_tty_or_raise_error(error_message)
             azure_rm_service_principal_key = prompt_pass('Azure RM service principal key:', confirm=True)
         else:
             azure_rm_service_principal_key = os.environ[AZURE_RM_SP_KEY_END_VARIABLE_NAME]
 
-        service_endpoint_authorization = EndpointAuthorization(
-            parameters={'tenantid': azure_rm_tenant_id,
-                        'serviceprincipalid': azure_rm_service_principal_id,
-                        'authenticationType': 'spnKey',
-                        'serviceprincipalkey': azure_rm_service_principal_key},
-            scheme=SERVICE_ENDPOINT_AUTHORIZATION_SERVICE_PRINCIPAL)
-        service_endpoint_data = {
-            'subscriptionId': azure_rm_subscription_id,
-            'subscriptionName': azure_rm_subscription_name,
-            'environment': 'AzureCloud',
-            'creationMode': 'Manual'
-        }
-        service_endpoint_to_create = ServiceEndpoint(
-            authorization=service_endpoint_authorization, data=service_endpoint_data,
-            name=name, type=SERVICE_ENDPOINT_TYPE_AZURE_RM, url='https://management.azure.com/')
+        service_endpoint_authorization.parameters['authenticationType'] = 'spnKey'
+        service_endpoint_authorization.parameters['serviceprincipalkey'] = azure_rm_service_principal_key
+    else:
+        with open(azure_rm_service_principal_certificate_path, "r") as f:
```

Comment:
tested with utf-8 and ansi.. let see if we can break this and then we can add option to pass encoding..  not adding it right now because if not needed than that will be unnecessary complication